docs/proposals/06-immutable-etcd-backups.md

```diff
@@ -0,0 +1,292 @@
+
+- **Immutability is Maintained:** New snapshots have a fresh immutability period, keeping backups protected.
+- **Operator Intentions are Respected:** The ETCD cluster remains in its hibernated state, as per the operator's intention.
+- **Storage Costs are Controlled:** Avoids unnecessary extension of immutability for all snapshots, preventing increased storage costs.
+
+### New Compaction Controller Flag
+
+**Flag:** `--hibernation-snapshot-interval`
+
+- **Type:** Duration
+- **Default:** `24h`
+- **Description:** This flag sets the period after which a compaction job should be triggered for a hibernated ETCD cluster, based on the time since the last renewal of the full snapshot lease. If the time since `fullLease.Spec.RenewTime.Time` exceeds the duration specified by this flag, and `etcd.spec.replicas` is `0` (indicating hibernation), the compaction job will automatically trigger to create a new snapshot. This approach ensures that backups remain within the immutability period and are safeguarded against becoming mutable.
+
+### Garbage Collection during Compaction for Hibernated ETCD Clusters
+
+If an ETCD cluster is hibernated for a long duration, there is a chance that backup storage will accumulate a large number of snapshots. Since the ETCD cluster is not running, the `etcd-backup-restore` component is also not running to perform garbage collection of the snapshots.
+
+To address this, the compaction job should be enhanced to handle garbage collection during hibernation. This ensures that old snapshots are cleaned up appropriately, considering the immutability constraints.
+
+### Excluding Snapshots Under Specific Circumstances
+
+Given that immutable backups cannot be deleted until the immutability period expires, there are scenarios, such as corrupted snapshots or other anomalies, where certain snapshots must be skipped during the restoration process. To facilitate this:
+
+- **Custom Metadata Tags:** Utilize custom metadata to mark specific objects (snapshots) that should be bypassed. To exclude a snapshot from the restoration process, attach custom metadata to it with the key `x-etcd-snapshot-exclude` and value `true`. This method is officially supported, as demonstrated in the [etcd-backup-restore PR](https://github.com/gardener/etcd-backup-restore/pull/776).
+
+## Compatibility
+
+The proposed changes are fully compatible with existing ETCD clusters and backup processes. Operators are responsible for creating or updating the immutability settings on the backup storage buckets, but no changes are required to the ETCD clusters themselves.
+
+- **Backward Compatibility:** Existing clusters without immutable buckets will continue to function without change.
+- **Forward Compatibility:** Clusters can opt-in to use immutable backups by configuring the bucket accordingly.
+
+## Implementation Steps
+
+1. **Enhance the Compaction Job:**
```

Comment:
I think we should create a new name for the job for Hibernated Full Snapshots and ensure that we have flags and even flow which can leverage the existing compaction feature and enhance it with additional change required for Immuatable backup snapshotting and garbage collection. 

Also we cannot have a compaction job for hibernated cluster in practical terms, so it will be even more confusing to see a compaction job running for a hibernated cluster.